src/python/pants/core/util_rules/external_tool.py

```diff
@@ -75,51 +75,8 @@ def decode(cls, version_str: str) -> ExternalToolVersion:
         return cls(version, platform, sha256, int(filesize))
 
 
-class ExternalTool(Subsystem, metaclass=ABCMeta):
-    """Configuration for an invocable tool that we download from an external source.
-
-    Subclass this to configure a specific tool.
-
-
-    Idiomatic use:
-
-    class MyExternalTool(ExternalTool):
-        options_scope = "my-external-tool"
-        default_version = "1.2.3"
-        default_known_versions = [
-          "1.2.3|linux_arm64 |feed6789feed6789feed6789feed6789feed6789feed6789feed6789feed6789|112233",
-          "1.2.3|linux_x86_64|cafebabacafebabacafebabacafebabacafebabacafebabacafebabacafebaba|878986",
-          "1.2.3|macos_arm64 |deadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeef|222222",
-          "1.2.3|macos_x86_64|1234abcd1234abcd1234abcd1234abcd1234abcd1234abcd1234abcd1234abcd|333333",
-        ]
-
-        version_constraints = ">=1.2.3, <2.0"
-
-        def generate_url(self, plat: Platform) -> str:
-            ...
-
-        def generate_exe(self, plat: Platform) -> str:
-            return "./path-to/binary
-
-    @rule
-    def my_rule(my_external_tool: MyExternalTool, platform: Platform) -> Foo:
-        downloaded_tool = await Get(
-            DownloadedExternalTool,
-            ExternalToolRequest,
-            my_external_tool.get_request(platform)
-        )
-        ...
-    """
-
-    # The default values for --version and --known-versions, and the supported versions.
-    # Subclasses must set appropriately.
-    default_version: str
-    default_known_versions: list[str]
-    version_constraints: str | None = None
-
-    def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
-        self.check_version_constraints()
+class ExternalToolOptionsMixin:
```

Comment:
I didn't find much precedence for this kind of code reuse, but I'm not sure how else to fit the square "downloaded external tool" peg into the bootstrapping hole, as the interfaces for `TemplatedExternalTool` just doesn't fit `semver` specs.